WNPRC_Purchasing/src/org/labkey/wnprc_purchasing/WNPRC_PurchasingController.java

```diff
@@ -146,6 +174,39 @@ public ModelAndView getView(Object o, BindException errors)
         public void addNavTrail(NavTree root) { }
     }
 
+    @RequiresPermission(InsertPermission.class)
+    public class GetFolderAdminsAction extends ReadOnlyApiAction
+    {
+        @Override
+        public Object execute(Object o, BindException errors) throws Exception
+        {
+            Map<String, Object> resultProperties = new HashMap<>();
+            try
+            {
+                List<User> folderAdmins = getFolderAdmins(); //this doesn't include site admins
+                JSONArray results = new JSONArray();
+
+                for (User user : folderAdmins)
+                {
+                    JSONObject json = new JSONObject();
+                    json.put("userId", user.getUserId());
+                    json.put("displayName", user.getDisplayName(getUser()));
+                    results.put(json);
+                }
+
+                resultProperties.put("success", true);
+                resultProperties.put("results", results);
+                return new ApiSimpleResponse(resultProperties);
+            }
+            catch (IllegalArgumentException e)
```

Comment:
When does IllegalArgumentException get thrown? In general, we shouldn't be catching and handling RuntimeExceptions - we should prevent them from being thrown in the first place.

Reply:
Removed